doc/Developing.md

```diff
@@ -0,0 +1,124 @@
+    $ sudo apt-get update
+    $ sudo apt-get install sbt
+    ```
+  * Running sbt for the first time takes several minutes, because it downloads all required packages.
+* We also recommend to install [Metals](https://scalameta.org/metals/), which enables better integration with your IDE
+  * For VSCode, install the "Scala (Metals)" and "Scala Syntax (official)" plugins
+* You will also need Python installed (we recommend version 3.6+) and dx-toolkit (`pip install dxpy`)
+
+## Getting the source code
+
+* Clone or fork the [wdlTools repository](https://github.com/dnanexus-rnd/wdlTools) (depending on whether you have commit permissions)
+* Checkout an existing branch or create a new branch (e.g. feat/42-my-feature)
+* Add pre-commit hooks:
+  * Create/edit a file .git/hooks/pre-commit
+  * Add the following lines
+    ```bash
+    #!/bin/bash
+    check=$(sbt scalafmtCheckAll)
+    if [[ "$?" -ne "0" ]]; then
+      echo "Reformatting; please commit again"
+      sbt scalafmtAll
+      exit $check
+    fi
+    ```
+  * Make the file executable (e.g. `chmod +x .git/hooks/pre-commit`)
+  * This hook runs the code formatter before committing code. You can run this command manually, but it is easiest just to have it run automatically.
+
+## Adding new code
+
+* Follow the style guidelines (below).
+* Always write unit tests for any new code you add, and update tests for any code you modify.
+  * Unit tests should assert, and not print to the console
+  * WDL test files belong in the top directory `test`
+* Submit a pull request when you are ready to have your code reviewed.
+
+### Style guidelines
+
+* We use [scalafmt style](https://scalameta.org/scalafmt/) with a few modifications. You don't need to worry so much about code style since you will use the automatic formatter on your code before it is committed.
+* Readability is more important than efficiency or concision - write more/slower code if it makes the code more readable.
+* Avoid using more complex features, e.g. reflection.
+
+## Using sbt
+
+sbt is the build system for Scala (similar to a Makefile). The following are the main commands you will use while developing.
+
+### Compiling the code
+
+Scala (like Java) is a compiled language. To compile, run:
+
+```
+$ sbt compile
+```
+
+If there are errors in your code, the compiler will fail with (hopefully useful) error messages.
+
+### Running unit tests
+
+You should always run the unit tests after every successful compile. Generally, you want to run `sbt testQuick`, which only runs the tests that failed previously, as well as the tests for any code you've modified since the last time you ran the tests. However, the first time you checkout the code (to make sure your development environment is set up correctly) and then right before you push any changes to the repository, you should run the full test suite using `sbt test`.
+
+### Running the integration tests
```

Comment:
Do we need to use `dx login` for anything in this repository? I was hoping we wouldn't.